src/System.Private.ServiceModel/src/System/IdentityModel/Claims/ClaimComparer.cs

```diff
@@ -258,12 +255,36 @@ int IEqualityComparer.GetHashCode(object obj)
             }
         }
 
-#if SUPPORTS_WINDOWSIDENTITY
+//#if SUPPORTS_WINDOWSIDENTITY
         private class UpnObjectComparer : IEqualityComparer
         {
             bool IEqualityComparer.Equals(object obj1, object obj2)
             {
-                throw ExceptionHelper.PlatformNotSupported();
+                if (StringComparer.OrdinalIgnoreCase.Equals(obj1 as string, obj2 as string))
+                    return true;
+
+                string upn1 = obj1 as string;
+                string upn2 = obj2 as string;
+                if (upn1 == null || upn2 == null)
+                    return false;
+
+#if SUPPORTS_WINDOWSIDENTITY
+                SecurityIdentifier sid1;
+                if (!TryLookupSidFromName(upn1, out sid1))
+                    return false;
+
+                // Normalize to sid
+                SecurityIdentifier sid2;
+                if (!TryLookupSidFromName(upn2, out sid2))
+                    return false;
+
+                return sid1 == sid2;
+#else
+                // If WindowsIdentity isn't supported, then we can't
+                // retrieve the SecurityIdentifier's to compare so
+                // must return false
+                return false;
```

Comment:
I guess this is going to be a necessary evil in order for UWP to work... we however need to clearly show this in the release notes.